wiki_articles/compilation_steps.md

```diff
@@ -0,0 +1,26 @@
+# How Is C/C++ Source Code Compiled?
+
+\:hash: The **preprocessor** expands macros and copy-pastes the content of included files
+at their point of inclusion. After being preprocessed, a source file no longer
+contains preprocessor directives. The result is called a _translation unit_ (TU).
+
+> :bulb: _The preprocessor merely does text replacement, it does not care about any aspect of C/C++._
+
+:ce: The **compiler** turns instructions into machine code and enforces C/C++
+syntax rules. In particular, anything that is used must be have been declared
+beforehand. The compiler only needs to know that something **exists**, not how
+it is _defined_.
+
+> :bulb: _Headers are a convenient way to expose function declarations to the compiler. Function definitions can reside in another, separately compiled TU, allowing finer control over the compilation process._
+
+As function definitions might be unavailable at this point, the compiler emits
+dummy machine code for function calls. A final step must be performed to fix those.
```

Comment:
Should these two lines go under the above 💡 blockquote? Happy to merge it either way

Reply:
Makes more sense that they do yes :+1: